src/KubernetesClient/Kubernetes.Generic.cs

```diff
@@ -0,0 +1,1131 @@
+                ex.Response = new HttpResponseMessageWrapper(httpResponse, responseContent);
+                if (shouldTrace)
+                {
+                    ServiceClientTracing.Error(invocationId, ex);
+                }
+                httpRequest.Dispose();
+                if (httpResponse != null)
+                {
+                    httpResponse.Dispose();
+                }
+                throw ex;
+            }
+            // Create Result
+            var result = (HttpOperationResponse)Activator.CreateInstance(typeof(HttpOperationResponse<>).MakeGenericType(type));
+            result.Request = httpRequest;
+            result.Response = httpResponse;
+            // Deserialize Response
+            if (statusCode == OK)
+            {
+                responseContent = await httpResponse.Content.ReadAsStringAsync().ConfigureAwait(false);
+                try
+                {
+                    result.GetType().GetProperty("Body").SetValue(result, DeserializeObject(type, responseContent));
+                }
+                catch (JsonException ex)
+                {
+                    httpRequest.Dispose();
+                    if (httpResponse != null)
+                    {
+                        httpResponse.Dispose();
+                    }
+                    throw new SerializationException("Unable to deserialize the response.", responseContent, ex);
+                }
+            }
+            if (shouldTrace)
+            {
+                ServiceClientTracing.Exit(invocationId, result);
+            }
+            return result;
+        }
+
+        /// <inheritdoc cref="IKubernetes"/>
+        public async Task<HttpOperationResponse<T>> ReadWithHttpMessagesAsync<T>(
+            string name,
+            string namespaceParameter = null,
+            bool? exact = default,
+            bool? export = default,
+            string pretty = default,
+            Dictionary<string, List<string>> customHeaders = null,
+            CancellationToken cancellationToken = default) where T : IKubernetesObject
+        {
+            var result = await ReadWithHttpMessagesAsync(typeof(T), name, namespaceParameter, exact, export, pretty, customHeaders, cancellationToken);
+            return (HttpOperationResponse<T>) result;
+        }
+
+        /// <inheritdoc cref="IKubernetes"/>
+        public async Task<HttpOperationResponse> ReadWithHttpMessagesAsync(
+            Type type,
+            string name,
+            string namespaceParameter = null,
+            bool? exact = default,
+            bool? export = default,
+            string pretty = default,
+            Dictionary<string, List<string>> customHeaders = null,
+            CancellationToken cancellationToken = default)
+        {
+
+            // Tracing
+            var shouldTrace = ServiceClientTracing.IsEnabled;
+            string invocationId = null;
+            if (shouldTrace)
```

Comment:
Though this logic is a little different than previous, I think you could pretty easily refactor it into a helper method that was something like:

```
if (shouldTrace)
{
    addTraces(["exact", exact], ["export", export], ....);
}